### PR TITLE
Update SPI detect test cases (Bugfix)

### PR DESCRIPTION
**Pull request by stanley31huang**
## Description
Modified the SPI detect test cases as validate the device nodes is match or not.

## Resolved issues
N/A

## Documentation
N/A

## Tests
side-loading test results:
https://certification.canonical.com/hardware/202405-33984/submission/386459/

### CODE REVIEW

File: contrib/checkbox-ce-oem/checkbox-provider-ce-oem/units/spi/jobs.pxu [hunk 1]
```diff
@@ -11,10 +11,14 @@ _summary:
     To detect if the spi device exist
 _purpose:
     Check if the SPI devices exist
+_description:
+    This test will validate the SPI devices are exposed as expected.
+    Update the SPI_DEVICES variable if you have multiple SPI devices. the default value is "/dev/spidev0.0"
+    e.g. SPI_DEVICES="/dev/spidev0.0 /dev/spidev0.1 /dev/spidev0.3"
```
Comment (LiaoU3) [inline contrib/checkbox-ce-oem/checkbox-provider-ce-oem/units/spi/jobs.pxu:17]: Do you think separate it by comma instead of space will be a good idea?

**Review by LiaoU3 — CHANGES_REQUESTED** [review]
Everything looks great except one question. Thank you!

File: contrib/checkbox-ce-oem/checkbox-provider-ce-oem/bin/spidev_test.py [hunk 2]
```diff
@@ -17,19 +17,20 @@ def runcmd(command):
...
+def detect_spi_node(expected_devs=None):
     spi_devices = glob.glob("/dev/spidev*")
     if spi_devices:
         for device in spi_devices:
             print("device: {}".format(device))
             print()
-    if exit_with_notice:
-        if len(spi_devices) != expected_number:
+    if expected_devs:
+        if set(spi_devices) != set(expected_devs.split()):
```
Comment (stanley31huang) [inline contrib/checkbox-ce-oem/checkbox-provider-ce-oem/bin/spidev_test.py:27]:
```suggestion
        if set(spi_devices) != set(expected_devs.split(",")):
```

File: contrib/checkbox-ce-oem/checkbox-provider-ce-oem/units/spi/jobs.pxu (hunk 1 above)
Comment (stanley31huang) [inline contrib/checkbox-ce-oem/checkbox-provider-ce-oem/units/spi/jobs.pxu:17]:
```suggestion
    e.g. SPI_DEVICES="/dev/spidev0.0,/dev/spidev0.1,/dev/spidev0.3"
```

Comment (stanley31huang) [reply contrib/checkbox-ce-oem/checkbox-provider-ce-oem/units/spi/jobs.pxu]: fixed

**Review by LiaoU3 — APPROVED** [review]
+1

**Pull request merged**